pkg/hierarchy/manager_test.go

```diff
@@ -374,6 +374,125 @@ func TestManager(t *testing.T) {
 	}
 }
 
+func TestCycles(t *testing.T) {
+	type M = Manager[*testClusterQueue, *testCohort]
+	cases := map[string]struct {
+		operations func(M)
+		wantCycles map[string]bool
+	}{
+		"no cycles": {
```

Comment:
Add a test case for a diamond structure. 
```
    root
  /        \
left     right
  \          /
 shared child
```

Reply:
We can't represent this configuration, as a child can only have one parent

Reply:
sure :)